proto/v1/org_policy_service.proto

```diff
@@ -54,6 +54,149 @@ service OrgPolicyService {
+      additional_bindings {
+        patch: "/v1/{policy.name=environments/*/policies/*}"
+        body: "policy"
+      }
+      additional_bindings {
+        patch: "/v1/{policy.name=environments/*/instances/*/policies/*}"
+        body: "policy"
+      }
+      additional_bindings {
+        patch: "/v1/{policy.name=environments/*/instances/*/databases/*/policies/*}"
+        body: "policy"
+      }
+    };
+    option (google.api.method_signature) = "policy,update_mask";
+  }
+
+  rpc DeletePolicy(DeletePolicyRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1/{name=policies/*}"
+
+      additional_bindings {
+        delete: "/v1/{name=projects/*/policies/*}"
+      }
+      additional_bindings {
+        delete: "/v1/{name=environments/*/policies/*}"
+      }
+      additional_bindings {
+        delete: "/v1/{name=environments/*/instances/*/policies/*}"
+      }
+      additional_bindings {
+        delete: "/v1/{name=environments/*/instances/*/databases/*/policies/*}"
+      }
+    };
+    option (google.api.method_signature) = "name";
+  }
+
+   rpc UndeletePolicy(UndeletePolicyRequest) returns (Policy) {
+    option (google.api.http) = {
+      post: "/v1/{name=policies/*}:undelete"
+      body: "*"
+
+      additional_bindings {
+        post: "/v1/{name=projects/*/policies/*}:undelete"
+        body: "*"
+      }
+      additional_bindings {
+        post: "/v1/{name=environments/*/policies/*}:undelete"
+        body: "*"
+      }
+      additional_bindings {
+        post: "/v1/{name=environments/*/instances/*/policies/*}:undelete"
+        body: "*"
+      }
+      additional_bindings {
+        post: "/v1/{name=environments/*/instances/*/databases/*/policies/*}:undelete"
+        body: "*"
+      }
+    };
+  }
+}
+
+message CreatePolicyRequest {
+  // The parent resource where this instance will be created.
+  // Workspace resource name: "".
+  // Environment resource name: environments/environment-id.
+  // Instance resource name: environments/environment-id/instances/instance-id.
+  // Database resource name: environments/environment-id/instances/instance-id/databases/database-name.
+  string parent = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // The policy to create.
+  Policy policy = 2 [(google.api.field_behavior) = REQUIRED];
+
+  PolicyType type = 3;
+}
+
+message UpdatePolicyRequest {
+  // The policy to update.
+  //
+  // The policy's `name` field is used to identify the instance to update.
+  // Format: {resource name}/policies/{policy type}
+  // Workspace resource name: "".
+  // Environment resource name: environments/environment-id.
+  // Instance resource name: environments/environment-id/instances/instance-id.
+  // Database resource name: environments/environment-id/instances/instance-id/databases/database-name.
+  Policy policy = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // The list of fields to update.
+  google.protobuf.FieldMask update_mask = 2;
```

Comment:
https://google.aip.dev/134

We can introduce allow_missing to provide upsert semantics. Also, the get should always return stuff, default policy when the policy doesn't exist in storage.

Reply:
> Also, the get should always return stuff, default policy when the policy doesn't exist in storage.

I guess also for listing policies?

Reply:
Done for allow_missing